htdocs/website/index.php

```diff
@@ -2247,13 +2248,91 @@
 
 
 /*
- * View
- */
+* View
+*/
 
 $form = new Form($db);
 $formadmin = new FormAdmin($db);
 $formwebsite = new FormWebsite($db);
 $formother = new FormOther($db);
+$domainname = '0.0.0.0:8080';
+$tempdir = $conf->website->dir_output.'/'.$websitekey.'/';
+
+// Confirm generation of website sitemaps
+if ($action == 'confirmgeneratesitemaps'){
+	$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"].'?website='.$website->ref, $langs->trans('ConfirmSitemapsCreation'), $langs->trans('ConfirmGenerateSitemaps', $object->ref), 'generatesitemaps', '', "yes", 1);
+	$action = 'preview';
+}
+
+// Generate web site sitemaps
+if ($action == 'generatesitemaps') {
```

Comment:
But this part is an action with no user interaction. result are stored with setEventMessages to be output later by the footer of page.
So this should be before the View (atfer the /* Action comment).